src/ui/public/state_management/__tests__/state.js

```diff
@@ -17,282 +17,322 @@ import StubBrowserStorage from 'test_utils/stub_browser_storage';
+          const { state, hashedItemStore, notifier } = setup({ storeInHash: true });
+          const fatalStub = sinon.stub(notifier, 'fatal').throws();
+          sinon.stub(hashedItemStore, 'setItem').returns(false);
+
+          expect(() => {
+            state.toQueryParam();
+          }).to.throwError();
 
-      it('presents fatal error linking to github when setting item fails', () => {
-        const { state, hashedItemStore, notifier } = setup({ storeInHash: true });
-        const fatalStub = sinon.stub(notifier, 'fatal').throws();
-        sinon.stub(hashedItemStore, 'setItem').returns(false);
+          sinon.assert.calledOnce(fatalStub);
+          expect(fatalStub.firstCall.args[0]).to.be.an(Error);
+          expect(fatalStub.firstCall.args[0].message).to.match(/github\.com/);
+        });
 
-        expect(() => {
-          state.toQueryParam();
-        }).to.throwError();
+        it('translateHashToRison should gracefully fallback if parameter can not be parsed', () => {
+          const { state, hashedItemStore } = setup({ storeInHash: false });
 
-        sinon.assert.calledOnce(fatalStub);
-        expect(fatalStub.firstCall.args[0]).to.be.an(Error);
-        expect(fatalStub.firstCall.args[0].message).to.match(/github\.com/);
+          expect(state.translateHashToRison('(a:b)')).to.be('(a:b)');
+          expect(state.translateHashToRison('')).to.be('');
+
+          const existingHash = createStateHash('{"a": "b"}', () => null);
+          hashedItemStore.setItem(existingHash, '{"a": "b"}');
+
+          const nonExistingHash = createStateHash('{"b": "c"}', () => null);
+
+          expect(state.translateHashToRison(existingHash)).to.be('(a:b)');
+          expect(state.translateHashToRison(nonExistingHash)).to.be('!n');
+        });
       });
+    });
+  });
 
-      it('translateHashToRison should gracefully fallback if parameter can not be parsed', () => {
-        const { state, hashedItemStore } = setup({ storeInHash: false });
+  describe('Disabled', () => {
+    let state;
+    let $location;
+    let $rootScope;
+    let readUrlSpy;
+
+    beforeEach(ngMock.module('kibana', function (stateManagementConfigProvider) {
+      stateManagementConfigProvider.disable();
+    }));
+    beforeEach(ngMock.inject(function (_$rootScope_, _$location_, Private, config) {
+      const State = Private(StateProvider);
+      $location = _$location_;
+      $rootScope = _$rootScope_;
+      Notifier.prototype._notifs.splice(0);
+
+      const setup = opts => {
+        const { param, initial, storeInHash } = (opts || {});
+        sinon.stub(config, 'get').withArgs('state:storeInSessionStorage').returns(!!storeInHash);
+        return new State(param, initial);
+      };
 
-        expect(state.translateHashToRison('(a:b)')).to.be('(a:b)');
-        expect(state.translateHashToRison('')).to.be('');
+      state = setup();
+      readUrlSpy = sinon.spy(state, '_readFromURL');
```

Comment:
__question:__ ugh spying on `private` method :/ Is it possible to check `$location.search()` or something like that to make sure URL hasn't been modified instead? Obviously I won't block on that if it's difficult or worse for some reason, so up to you.

Reply:
That's not a bad idea. I'll give that a shot.